### PR TITLE
gh-104050: Argument Clinic: Annotate the BufferSeries class

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-104050

### CODE REVIEW

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -1981,11 +1984,11 @@ def __getitem__(self, i):
             self._array.append(self._constructor())
         return self._array[i]
 
-    def clear(self):
+    def clear(self) -> None:
         for ta in self._array:
-            ta._text.clear()
+            ta.text.clear()
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py:1986]: Oh, was this a bug? Should we add a regression test?

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:1986]: Yes, it was. I'll try to produce a test that triggers it.

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -1965,12 +1965,15 @@ class BufferSeries:
...
         self._start = 0
-        self._array = []
+        self._array: list[_TextAccumulator] = []
         self._constructor = _text_accumulator
 
-    def __getitem__(self, i):
+    def __getitem__(
+            self,
+            i: int
+    ) -> _TextAccumulator:
```
Comment (serhiy-storchaka) [inline Tools/clinic/clinic.py]: Why not write it in one line? It is still pretty short.

**Review by serhiy-storchaka — APPROVED** [review]

File: Tools/clinic/clinic.py (hunk 2 above)
Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Sure; I have not yet developed a preference when it comes to annotation style :)

Comment (AlexWaygood) [reply Tools/clinic/clinic.py]: FWIW, I agree with Serhiy — I would probably put this all on one line — but I also don't really care much either way (life's too short to spend it arguing over code formatting preferences 😄)

Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: > (life's too short to spend it arguing over code formatting preferences 😄)

No, it isn't! 😆

**Comment by erlend-aasland** [comment]
On hold until #106972 has landed.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Comment by erlend-aasland** [comment]
Thanks for the reviews!

**Pull request merged**